### PR TITLE
[11.0][auditlog] Logging `create` events on `account.move.line` breaks invoice validation and Journal Entry validation seperately

### PR DESCRIPTION
**Pull request by smangukiya**
similar ref issue #1134

### CODE REVIEW

File: auditlog/models/rule.py [hunk 1]
```diff
@@ -255,6 +263,7 @@ def read(self, fields=None, load='_classic_read', **kwargs):
                 self.env.uid, self._name, self.ids,
                 'read', read_values, None, {'log_type': log_type})
             return result
+
```
Comment (sebalix) [inline auditlog/models/rule.py]: No need to add a blank line here, not relevant for the fix.

File: auditlog/models/rule.py [hunk 2]
```diff
@@ -269,11 +278,11 @@ def write_full(self, vals, **kwargs):
             rule_model = self.env['auditlog.rule']
             old_values = dict(
                 (d['id'], d) for d in self.sudo()
-                .with_context(prefetch_fields=False).read(list(self._fields)))
+                    .with_context(prefetch_fields=False).read(list(self._fields)))
```
Comment (sebalix) [inline auditlog/models/rule.py]: No need to indent (and it adds a lint error on Travis).

File: auditlog/models/rule.py [hunk 3]
```diff
@@ -269,11 +278,11 @@ def write_full(self, vals, **kwargs):
...
             result = write_full.origin(self, vals, **kwargs)
             new_values = dict(
                 (d['id'], d) for d in self.sudo()
-                .with_context(prefetch_fields=False).read(list(self._fields)))
+                    .with_context(prefetch_fields=False).read(list(self._fields)))
```
Comment (sebalix) [inline auditlog/models/rule.py]: No need to indent (and it adds a lint error on Travis).

File: auditlog/models/rule.py [hunk 4]
```diff
@@ -310,7 +319,7 @@ def unlink_full(self, **kwargs):
             rule_model = self.env['auditlog.rule']
             old_values = dict(
                 (d['id'], d) for d in self.sudo()
-                .with_context(prefetch_fields=False).read(list(self._fields)))
+                    .with_context(prefetch_fields=False).read(list(self._fields)))
```
Comment (sebalix) [inline auditlog/models/rule.py]: No need to indent (and it adds a lint error on Travis).

File: auditlog/models/rule.py [hunk 5]
```diff
@@ -536,3 +555,4 @@ def unsubscribe(self):
                 act_window.unlink()
         self.write({'state': 'draft'})
         return True
+
```
Comment (sebalix) [inline auditlog/models/rule.py]: Remove this empty line.

**Review by sebalix — CHANGES_REQUESTED** [review]
Hi, thank you for trying to fix this issue.
Is it possible to add a test which reproduces the initial problem?

**Comment by smangukiya** [comment]
@sebalix : Hi, Resolved all code formatting related changes. Thanks.

**Review by max3903 — APPROVED** [review]

**Comment by smangukiya** [comment]
@max3903 @sebalix : Reminder. Can we please merge this PR now? Thanks

**Comment by max3903** [comment]
@smangukiya : @sebalix requested a test to catch this scenario.

**Comment by sebalix** [comment]
@smangukiya I've made a fix which is a bit cleaner I think: https://github.com/OCA/server-tools/pull/1686, base on the record cache (almost like you) and using `convert_to_read` from fields to get the expected value format (like `read`).

**Comment by sebalix** [comment]
@pedrobaeza @StefanRijnhart can be closed, #1686 has been merged in favor of this one

**Comment by StefanRijnhart** [comment]
@smangukiya I'm closing this per the last comment. Thank you for your contribution. I'm sorry it was not merged but I'm sure it contributed to the eventual solution!

**Pull request closed without merging**